### DISCUSSION TITLE
Multichain Apps

### DISCUSSION THREAD

**Discussion by Zimtente** [wevm/wagmi#4109] [General]
hey there,

my question is: what is the best approach for "chain switching"? Is there anything build in? 

If we supply "chainId" for the useWriteContract hook it actually just throws an error if not connected with the correct chain. We atm have our own hook that does both writing and switching if not connected with the correct chain - though had some issues with:

1) coinbase wallet on mobile race conditions
2) wagmi useAccount being stale and not having the correct chain (we now use useWalletClient as we can refetch it)

so wondered if there is a "buildin" way of handling chain switching.

**Comment by tmm** [comment]
`useAccount` shouldn't be stale. Feel free to create a bug if you think there is an issue!
